hil/ext/switches/dellnos9.py

```diff
@@ -0,0 +1,387 @@
+
+        Args:
+            interface: interface to add the vlan to
+            vlan: vlan to add
+        """
+        if not self._is_port_on(interface):
+            self._port_on(interface)
+        url = self._construct_url()
+        command = 'interface vlan ' + vlan + '\r\n tagged ' + \
+            self.interface_type + ' ' + interface
+        payload = self._make_payload(CONFIG, command)
+        self._make_request('POST', url, data=payload)
+
+    def _remove_vlan_from_trunk(self, interface, vlan):
+        """ Remove a vlan from a trunk port.
+
+        Args:
+            interface: interface to remove the vlan from
+            vlan: vlan to remove
+        """
+        url = self._construct_url()
+        command = 'interface vlan ' + vlan + '\r\n no tagged ' + \
+            self.interface_type + ' ' + interface
+        payload = self._make_payload(CONFIG, command)
+        self._make_request('POST', url, data=payload)
+
+    def _remove_all_vlans_from_trunk(self, interface):
+        """ Remove all vlan from a trunk port.
+
+        Args:
+            interface: interface to remove the vlan from
+
+        this is kind slow for a number of reasons
+        1. we are using the REST API CLI
+        2. we have to go to every vlan and then remove our port from it;
+           that makes multiple calls.
+        """
+        for vlan in self._get_vlans(interface):
+            self._remove_vlan_from_trunk(interface, vlan[1])
+
+    def _set_native_vlan(self, interface, vlan):
+        """ Set the native vlan of an interface.
+
+        Args:
+            interface: interface to set the native vlan to
+            vlan: vlan to set as the native vlan
+
+        Method relies on the REST API CLI which is slow
+        """
+        if not self._is_port_on(interface):
+            self._port_on(interface)
+        url = self._construct_url()
+        command = 'interface vlan ' + vlan + '\r\n untagged ' + \
+            self.interface_type + ' ' + interface
+        payload = self._make_payload(CONFIG, command)
+        self._make_request('POST', url, data=payload)
+
+    def _remove_native_vlan(self, interface):
+        """ Remove the native vlan from an interface.
+
+        Args:
+            interface: interface to remove the native vlan from.vlan
+
+        Method relies on the REST API CLI which is slow.
+        """
+        vlan = self._get_native_vlan(interface)[1]
+        url = self._construct_url()
+        command = 'interface vlan ' + vlan + '\r\n no untagged ' + \
+            self.interface_type + ' ' + interface
+        payload = self._make_payload(CONFIG, command)
+        self._make_request('POST', url, data=payload)
+
+    def _port_shutdown(self, interface):
+        """ Shuts down <interface>
+
+        Turn off portmode hybrid, disable switchport, and then shut down the
+        port. All non-default vlans must be removed before calling this.
+
+        Fast method that uses the REST API (no hackery).
```

Comment:
This doesn't really belong in the docstring, as it's more of an implementation question (same goes elsewhere). If you want to include it, I would do it in a comment just below this, but what I think makes more sense is to put TODO/XXX comments int he places where improvement is desired.